variables.tf

```diff
@@ -46,7 +46,14 @@ variable "retention" {
   default     = 30
 }
 
+variable "provision_trigger" {
+  type        = bool
+  description = "Whether or not to create a lambda trigger from an SNS topic"
+  default     = "false"
+}
+
 variable "sns_topic_arn" {
-  type        = string
+  type        = list(string)
   description = "SNS Topic ARN"
+  default     = ["undefined"]
```

Comment:
Instead of having a provision_trigger boolean could the default on this just be blank list?  Would the length(va.sns_topic_arn) check then return 0?

Reply:
If I don't set a default value, it considers it a required variable and errors out if it is not defined. From what I have read it's not possible to set an empty list in TF. This is the least hacky solution I could come up with.

Reply:
I meant trying: `default     = []`

Reply:
An empty list creates other problems that have to be solved in other ways. If you try to access the length of an empty list you will get an error. The boolean seemed to be the most elegant way to work around that issue.